### DISCUSSION TITLE
Using bash builtins and while loops to launch in a tab

### DISCUSSION THREAD

**Discussion by Strykar** [kovidgoyal/kitty#4821] [Q&A] [answered]
kitty is auto started on boot via XDG-autostart and opens a few tabs that run programs based on if certain services have started successfully. Till today, I just used `sleep 60` to give the services enough time before starting my programs.

Today I tried `while systemctl status livepeer; do journalctl --catalog --boot --follow --unit livepeer; done` but that shows the output of the `systemctl status` command and does not execute the `do journalctl` when I try to invoke it via a session file like `bash -c 'while systemctl status livepeer; do journalctl --catalog --boot --follow --unit livepeer; done'`

What am I missing here?

**Comment by page-down** [comment, marked as answer]
I tried and it works fine.
Maybe you need `systemctl is-active --quiet service-here`.

```conf
# startup-session.conf
launch --allow-remote-control bash -c "printf waiting; while test ! -f /tmp/flag >/dev/null 2>&1; do printf .; sleep 1; done; kitty @ launch sh -c 'echo ready; exec bash -il'"
```

```shell
kitty --session ./path/to/startup-session.conf
touch /tmp/flag
```

**Comment by Strykar** [comment]
@page-down Thank you! :heart: 
`launch bash -c 'while systemctl is-active --quiet livepeer; do livepeer_cli; done'` works!

Any idea why it would fail when using `systemctl status`?

**Reply by page-down** [reply]
This is because the command will not exit.
You need `systemctl status --lines=0 service-here `.

If you want to keep the log displayed at startup, end the log viewer when service is up, and start the command line tool.
Then you need to write some more scripts.

**Reply by Strykar** [reply]
Ah, perhaps `--no-pager` might make it exit too? Thanks, that helps!
